cloud/services/bastionhosts/bastionhosts.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bastionhosts
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/converters"
+)
+
+// Spec specification for bastion host
+type Spec struct {
+	Name         string
+	SubnetName   string
+	PublicIPName string
+	VnetName     string
+}
+
+// Get provides information about a bastion host.
+func (s *Service) Get(ctx context.Context, spec interface{}) (interface{}, error) {
```

Comment:
please rebase + remove the `Get()` to align with https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/713

Reply:
removed the `Get()` @CecileRobertMichon